incubator/declarative-linking/src/main/java/org/glassfish/jersey/linking/EntityDescriptor.java

```diff
@@ -190,4 +212,44 @@ private void findLinkHeaders(Class<?> entityClass) {
             findLinkHeaders(ic);
         }
     }
+    
+    /**
+     * Find and cache the methods of the supplied class and its superclasses and
+     * interfaces.
+     * @param entityClass the class
+     */
+    private void findMethods(Class<?> entityClass) {
+        for (Method m: entityClass.getMethods()) {
+            InjectLink a = m.getAnnotation(InjectLink.class);
```

Comment:
Would it be possible to create a more general helper method, that would contain some common shared code? Basically from here to the end of loop, this is almost identical with `findFileds` (only the "output collection"is different), which is 100% understandable - it does the same thing as `findFields`, just for methods (except it doesn't need to look into superclasses and interfaces).
Just a suggestion, if it would make the code less readable, we can leave this way (after all, this is still in incubator).